### PR TITLE
[report,Plugin] Enumerate network devices for plugins

### PR DESCRIPTION
**Pull request by TurboTurtle**
This patchset first adds enumeration of network devices to the `devices` dict handed to plugins for device iteration. Second, it updates `add_device_cmd()` to allow handling of the `subdir` parameter, and finally it updates various network-related plugins to use `add_device_cmd()` instead of enumerating devices themselves.

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

**Comment by TurboTurtle** [comment]
Hmmm...it's strange that centos is getting a new kmod loaded now (team) only on one test...especially when there are no `team` devices on the test VMs...

File: sos/report/__init__.py [hunk 1]
```diff
@@ -522,6 +524,131 @@ def _get_namespaces(self):
...
+                dev for dev in listdir('/sys/class/net', self.opts.sysroot)
+                if dev not in ['bonding_masters']
+            ]
+        else:
+            try:
+                _nscmd = "ip netns exec %s ls /sys/class/net" % namespace
+                _nsout = sos_get_command_output(_nscmd)
+                if _nsout['status'] == 0:
+                    for _nseth in _nsout['output'].split():
+                        _eth_devs.append(_nseth)
```
Comment (pmoravec) [inline sos/report/__init__.py:617]: Devices in a namespace can't be `bonding_masters` (cf. line 609)?

File: sos/report/__init__.py [hunk 2]
```diff
@@ -522,6 +524,131 @@ def _get_namespaces(self):
...
+
+        :returns:   All valid ethernet devices found, potentially within a
+                    given namespace
+        :rtype:     ``list``
+        """
+        _eth_devs = []
+        if not namespace:
+            _eth_devs = [
+                dev for dev in listdir('/sys/class/net', self.opts.sysroot)
+                if dev not in ['bonding_masters']
```
Comment (pmoravec) [inline sos/report/__init__.py:609]: Isn't `if dev != 'bonding_master'` easier comparison?

File: sos/report/plugins/networking.py [hunk 3]
```diff
@@ -141,33 +141,28 @@ def setup(self):
...
+        ], devices='ethernet')
+
+        self.add_device_cmd([
+            "ethtool %(dev)s",
+            "ethtool --phy-statistics %(dev)s",
+            "ethtool --show-priv-flags %(dev)s",
+            "ethtool --show-eee %(dev)s",
+            "tc -s filter show dev %(dev)s",
+            "tc -s filter show dev %(dev)s ingress",
+        ], devices="ethernet")
```
Comment (pmoravec) [inline sos/report/plugins/networking.py:156]: .. and the (here) missing `tags=eth` is added automatically via `_add_device_cmd default method option `tags=_dev_tags`, so no "regression" :)

**Comment by pmoravec** [comment]
Just the two very minor comments, otherwise great!

File: sos/report/__init__.py (hunk 2 above)
Comment (TurboTurtle) [reply sos/report/__init__.py:609]: Initial thinking was that we'd have a handful of skip patterns, or at least one that might build up over time.

**Pull request merged**